internal/collector/otel_collector_plugin.go

```diff
@@ -225,7 +232,38 @@ func (oc *Collector) handleNginxConfigUpdate(ctx context.Context, msg *bus.Messa
 	}
 }
 
+func (oc *Collector) handleResourceUpdate(ctx context.Context, msg *bus.Message) {
+	resourceUpdateContext, ok := msg.Data.(*v1.Resource)
+	if !ok {
+		slog.ErrorContext(ctx, "Unable to cast message payload to *v1.Resource", "payload", msg.Data)
+		return
+	}
+
+	if oc.config.Collector.Processors.Attribute != nil { // attribute processor is enabled
```

Comment:
If `Attribute` is nil then we should create an `Attribute` object and add the action.
We should also check the slice of actions to see if there is already an action for "resource.id" or not